sdk/python/kfp/v2/compiler_cli_tests/compiler_cli_tests.py

```diff
@@ -23,24 +23,33 @@
 
 class CompilerCliTests(unittest.TestCase):
 
-  def _test_compile_py_to_json(self, file_base_name, additional_arguments=[]):
+  def _test_compile_py_to_json(self,
+                               file_base_name,
+                               additional_arguments=[],
+                               update_golden: bool = False):
```

Comment:
JFYI, I usually just execute the test pipeline `.py` file to update golden. This works for all recent samples:

https://github.com/kubeflow/pipelines/blob/4c51b576f577e337049599cb85b098006664265f/sdk/python/kfp/v2/compiler_cli_tests/test_data/pipeline_with_condition.py#L46-L50

Earlier samples have `output_path=__file__ + '.json'` instead, so I could take a look a the updated golden before renaming it to replace the existing one. But later I found that extra step is not really necessary.

Reply:
I see, got it, thanks!